es/src/main/scala/monix/connect/es/ElasticsearchSource.scala

```diff
@@ -0,0 +1,95 @@
+package monix.connect.es
+
+import com.sksamuel.elastic4s.requests.searches.{SearchHit, SearchRequest, SearchResponse}
+import com.sksamuel.elastic4s.{ElasticClient, RequestFailure, RequestSuccess, Response}
+import monix.eval.Task
+import monix.execution.internal.InternalApi
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.collection.mutable
+
+/**
+  *  A pre-built [[monix.reactive.Observable]] implementation that
+  *  publishes documents using an elasticsearch scroll cursor.
+  *
+  * @param request [[SearchRequest]]
+  * @param client an implicit instance of a [[ElasticClient]]
+  */
+@InternalApi private[es] class ElasticsearchSource(request: SearchRequest)(implicit client: ElasticClient)
+  extends Observable[SearchHit] {
+
+  import com.sksamuel.elastic4s.ElasticDsl._
+
+  private var scrollId: String = _
+  // Parse the keep alive setting out of the original query.
+  private val keepAlive = request.keepAlive.getOrElse("1m")
+
+  override def unsafeSubscribeFn(subscriber: Subscriber[SearchHit]): Cancelable = {
+    runLoop(mutable.Queue.empty, subscriber).runToFuture(subscriber.scheduler)
+  }
+
+  private def runLoop(buffer: mutable.Queue[SearchHit], sub: Subscriber[SearchHit]): Task[Unit] = {
```

Comment:
don't really like dealing with mutable variables... thought I think this implementation is good and safe

what do you think on merging run loop and fast loop together?

like:
```
  private def fastLoop(buffer: mutable.Queue[SearchHit], sub: Subscriber[SearchHit]): Task[Unit] = {
    for {
      _ <- fetch(buffer, sub)
      ack <- {
        if (buffer.nonEmpty) Task.deferFuture(sub.onNext(buffer.dequeue))
        else Task.now(Ack.Stop)
      }
    } yield {
      ack match {
        case Ack.Continue =>
          fastLoop(buffer, sub)
        case Ack.Stop =>
          sub.onComplete()
          Task.unit
      }
    }
  } 
```

Reply:
OK